### PR TITLE
Fix bugs with transport-ws-protocol

### PR DESCRIPTION
**Pull request by ayys**
I am guessing the `v3` branch was only used with `GraphQLWSHandler`, and not with `GraphQLTransportWSHandler`. That's probably why there were a bunch of bugs. This PR fixes the errors I got when trying to use subscriptions with the `graphql-transport-ws` protocol.

The first error I got was
```
Can't instantiate abstract class GraphQLTransportWSHandler with abstract method send_xjson
```
I implemented this function and the error went away.

Second error was with message types - they were `TypedDict`s, but typeddicts are essentially dicts.
This means additional fields specified are not properly used. I fixed it by converting them to `dataclasses` and adding a `to_dict` field to it.

### CODE REVIEW

File: graphql_server/channels/graphql_transport_ws.py [hunk 1]
```diff
@@ -24,10 +24,10 @@ def __init__(
         self._ws = ws
 
     async def get_context(self) -> Any:
-        return await self._get_context(request=self._ws)
+        return await self._get_context(self._ws)
 
     async def get_root_value(self) -> Any:
-        return await self._get_root_value(request=self._ws)
+        return await self._get_root_value(self._ws)
```
Comment (ayys) [inline graphql_server/channels/graphql_transport_ws.py:30]: Removed this because [channels/graphql_ws.py](https://github.com/graphql-python/graphql-server/blob/v3/graphql_server/channels/graphql_ws.py#L27) doesn't specify the argument name either.

File: graphql_server/channels/graphql_transport_ws.py [hunk 2]
```diff
@@ -37,9 +37,7 @@ async def close(self, code: int = 1000, reason: Optional[str] = None) -> None:
         await self._ws.close(code=code)
 
     async def handle_request(self) -> Any:
-        await self._ws.accept(
-            subprotocol=BaseGraphQLTransportWSHandler.GRAPHQL_TRANSPORT_WS_PROTOCOL
-        )
+        await self._ws.accept(subprotocol=BaseGraphQLTransportWSHandler.PROTOCOL)
```
Comment (ayys) [inline graphql_server/channels/graphql_transport_ws.py:40]: wrong variable name was being used :/

File: graphql_server/websockets/transport_ws_protocol/handlers.py [hunk 3]
```diff
@@ -210,8 +212,12 @@ async def handle_complete(self, message: CompleteMessage) -> None:
...
         await self.close(code=4400, reason=error_message)
 
-    async def send_message(self, data: TypedDict) -> None:
-        await self.send_message({**data, "type": data.type})
+    async def send_message(self, data: Message) -> None:
+        data = data.asdict()
+        assert (
+            data.get("type") is not None
+        ), "expected dict with `type` field. Got {} instead".format(data)
+        await self.send_json(data)
```
Comment (ayys) [inline graphql_server/websockets/transport_ws_protocol/handlers.py:220]: Use dataclasses instead of TypedDicts. `TypeDict` is just a dict at runtime, and is only used for typehinting. That means `data.type` would not have worked.
sidenote: we should do the same for `graphql_ws.py` too.

File: graphql_server/websockets/transport_ws_protocol/types.py [hunk 4]
```diff
@@ -5,6 +5,8 @@
 except ImportError:
     from typing_extensions import TypedDict
 
+from dataclasses import dataclass, asdict
+
 from .contstants import (
     GQL_CONNECTION_INIT,
     GQL_CONNECTION_ACK,
```
Comment (ayys) [inline graphql_server/websockets/transport_ws_protocol/types.py:12]: These constants should probably be an ENUM, but fixing that shouldn't be part of this PR

**Review by syrusakbary — APPROVED** [review]
